common/plugins/org.jboss.tools.common.jdt.debug/src/org/jboss/tools/common/jdt/debug/tools/ToolsCore.java

```diff
@@ -95,7 +96,17 @@ public Object getMonitoredVM() {
 	}
 	
 	public static boolean processIsRunning(String hostname, int vmPid) throws ToolsCoreException {
-		return Tools.getInstance().getMonitoredVm(hostname, vmPid) != null;
+		// If these throw exceptions, we want them to be rethrown
+		Object host = Tools.getInstance().invokeGetMonitoredHost(hostname);
+		Object vmId = Tools.getInstance().invokeVmIdentifier(String.format(IToolsConstants.VM_IDENTIFIRER, vmPid));
+		
+		// We want to catch and ignore these failures and just return that the process is not running
+		try {
+			Object monitoredVm = Tools.getInstance().invokeGetMonitoredVm(host, vmId);
+			return monitoredVm != null;
+		} catch(ToolsCoreException tce) {
```

Comment:
I didn't track the code that far, but it is always better to catch specific exceptions instead of generic ones. I see it is common approach in jdt.debug which IMO should be fixed as different issue.
For this pull request I would add new method processIsMonitorable() as you suggested which would explain why possible exceptions are ignored.